### ISSUE TITLE
Display missing JSON schema fields

### ISSUE THREAD

**Issue by ofrebourg** [josdejong/jsoneditor#446]
At the moment, if you have a schema containing more fields than the json object has, the missing fields do not appear in the tree view. You get a validation error on the root but the missing field itself does not appear.
![image](https://user-images.githubusercontent.com/6222281/29093685-8e66c74c-7c82-11e7-8560-42c7296e45fc.png)

![image](https://user-images.githubusercontent.com/6222281/29093663-7785107e-7c82-11e7-9405-830639f9deda.png)

It would be a great feature to be able to display all missing fields. In this case the value of such a field could be empty, which would show a placeholder value.
Diffing/merging the json and the schema could be outside of jsoneditor, but it would be a useful feature to have in jsoneditor.

**Comment by josdejong** [comment]
Thanks for your suggestion, I think this hits a quite common use case: having to manually add the missing fields that the schema is complaining about. 

So far, JSON Editor only displays your JSON for you, and allows manipulating it yourself. And via JSON schema it can warn you about issues. I don't think JSON Editor should _automatically_ change your JSON by adding missing fields or something, I think that would be confusing. But it would be great if we have an action which allow you to "fix" your JSON as much as possible according to the JSON Schema by simply clicking a button. I'm not sure what would be the best place for such a button "add missing fields". Maybe we could extend the popovers displaying JSON Schema issues with a button to fix the concerning issue. Or we could extend the context menu with a button.

**Comment by ofrebourg** [comment]
I agree, fixing the JSON doesn't need to be automatic. But setting an option or providing a way to programmatically fix the JSON would be a great feature. For me personally, that would be more useful than clicking a button, but other users might prefer clicking a button.

**Comment by josdejong** [comment]
Indeed. I think the desired behavior may differ for different applications. It's up to the developer for example to create a "skeleton" JSON document based on a JSON Schema. It will be nice to aid both developer and end users to easily create required fields.
